plugins/bulk-import/src/components/Repositories/DeleteRepositoryDialog.tsx

```diff
@@ -69,27 +70,33 @@ const DeleteRepositoryDialog = ({
   closeDialog: () => void;
 }) => {
   const classes = useStyles();
-  const [error, setError] = React.useState<string>('');
-  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [deleteError, setDeleteError] = React.useState<string>('');
   const bulkImportApi = useApi(bulkImportApiRef);
+  const deleteRepository = async (deleteRepo: AddRepositoryData) => {
+    return await bulkImportApi.deleteImportAction(
+      deleteRepo.repoUrl || '',
+      deleteRepo.defaultBranch || 'main',
+    );
+  };
+  const mutationDelete = useMutation(deleteRepository, {
+    onSuccess: (data: ImportJobStatus | Response) => {
+      if (get(data, 'err')) {
+        setDeleteError(`Unable to remove repository. ${get(data, 'err')}`);
+      } else {
+        closeDialog();
+      }
+    },
+    onError: (error: Error) => {
+      setDeleteError(`Unable to remove repository. ${error.message}`);
+    },
+  });
   const handleClickRemove = async () => {
-    setIsSubmitting(true);
     if (!repository.repoUrl || !repository?.defaultBranch) {
-      setIsSubmitting(false);
-      setError(
-        `Unable to remove repository. ${!repository?.repoUrl ? 'Repository URL missing' : 'Repository default branch is missing'}`,
+      setDeleteError(
+        `Unable to remove repository. ${!repository?.repoUrl ? 'Repository URL missing.' : 'Repository default branch is missing.'}`,
```

Comment:
You know that case before the user press on delete, right?

Then you can also show a static alert below and disable the error button in that case.

```tsx
const missingRepository = !repository.repoUrl || !repository?.defaultBranch;

{ missingRepository
  <Alert /> :
  null}
```